File: 0_basics/00_arrays/000_strings/001/solutions/solution.php
<?php
/**
 * @file solution.php
 * @author Mihail Croitor <mcroitor.gmail.com>
 * @brief `substring` task solution
 * @version 0.1
 */

$S = trim(fgets(STDIN));
list($POS, $N) = explode(' ', trim(fgets(STDIN)));
if ($N > strlen($S) - $POS) {
    $N = strlen($S) - $POS;
}
$substring = substr($S, $POS, $N);
echo $substring;

File: 0_basics/00_arrays/000_strings/001/solutions/solution.c
/**
 * @file solution.c
 * @author Mihail Croitor
 * @brief `Substring` task solution
 * @version 0.1
 * @date 2025-12-19
 * 
 * @copyright Copyright (c) 2025
 * 
 */

#include <stdio.h>

int main() {
    char S[1001];
    fgets(S, sizeof(S), stdin);
    int POS, N;
    scanf("%d %d", &POS, &N);
    for (int i = POS; i < POS + N && S[i] != '\0'; i++) {
        putchar(S[i]);
    }
    putchar('\n');
    return 0;
}